AEPCore/Sources/eventhub/EventHub.swift

```diff
@@ -45,15 +45,12 @@ final class EventHub {
         registerExtension(EventHubPlaceholderExtension.self, completion: { _ in })
 
         // Setup eventQueue handler for the main OperationOrderer
-        eventQueue.setHandler { (event) -> Bool in
-
-            let processedEvent = self.preprocessors.shallowCopy.reduce(event) { event, preprocessor in
-                preprocessor(event)
-            }
+        eventQueue.setHandler { [weak self] (event) -> Bool in
+            guard let processedEvent = self?.preprocessors.shallowCopy.reduce(event, {$1($0)}) else { return true }
```

Comment:
Rather than use an optional self (to avoid the nil-coalescing) I think unwrapping self would be best here

`guard let self = self else { return }`

Reply:
I think this suggestion is probably fine, but I am slightly concerned that with the suggestion, we're still just holding a strong reference to self which could result in the same issue we're seeing. I'd _slightly_ prefer to just weakly reference self.

Thoughts?

Reply:
Hmmm. I guess I've never fully questioned the weak-strong dance which has become the standard in these use cases. Let me think about it a bit more. But I'm cool with this going in. 👍🏽